src/askvm/resources/core/__test__/query.test.ts

```diff
@@ -0,0 +1,93 @@
+import { runUntyped } from '../../..';
+import { parse } from '../../../../askcode';
+import * as core from '../../core';
+import { query } from '../../core';
+
+const values = {
+  objectValue: {
+    x: [1, 2, 3, 4],
+  },
+  stringValue: 'string',
+  numberValue: 5,
+  booleanValue: true,
+  arrayVal: [1, 2, 3, 4],
+};
+
+function ask(code: string) {
+  return runUntyped(
+    {
+      resources: { ...core, query },
+      values,
+    },
+    parse(code)
+  );
+}
+
+describe('hasKey', () => {
+  it(`should return an array value`, async function () {
+    const expectedValueForArray = {
+      objectReturned: values.arrayVal,
+    };
+
+    const inLineArray = [1, 2, 3, 4];
+    const expectedValueForInlineArray = {
+      objectReturned: inLineArray,
+    };
+
+    const queryForAnArrayValue = `ask(query(node('objectReturned',f(call(get('get'),'arrayVal')))))`;
+    const queryForInlineArray = `ask(query(node('objectReturned',f(list(${inLineArray.join(
+      ','
+    )})))))`;
+
+    await expect(ask(queryForAnArrayValue)).resolves.toStrictEqual(
+      expectedValueForArray
+    );
+
+    await expect(ask(queryForInlineArray)).resolves.toStrictEqual(
+      expectedValueForInlineArray
+    );
+  });
+
+  it(`should return a string value`, async function () {
+    const expectedValue = {
+      objectReturned: values.stringValue,
+    };
+
+    const queryForAString = `ask(query(node('objectReturned',f(call(get('get'),'stringValue')))))`;
+
+    await expect(ask(queryForAString)).resolves.toStrictEqual(expectedValue);
+  });
+
+  it(`should return a number value`, async function () {
+    const expectedValue = {
+      objectReturned: values.numberValue,
+    };
+
+    const queryForANumber = `ask(query(node('objectReturned',f(call(get('get'),'numberValue')))))`;
+
+    await expect(ask(queryForANumber)).resolves.toStrictEqual(expectedValue);
+  });
+
+  it(`should return a boolean value`, async function () {
+    const expectedValue = {
+      objectReturned: values.booleanValue,
+    };
+
+    const queryForABoolean = `ask(query(node('objectReturned',f(call(get('get'),'booleanValue')))))`;
+
+    await expect(ask(queryForABoolean)).resolves.toStrictEqual(expectedValue);
+  });
+
+  // TODO :: should we make this work?
+  // it(`should return an object`, async function () {
+  //   const expectedObjectValue = {
+  //     objectReturned: values.objectValue,
+  //   };
+  //
+  //   const queryForAnObjectValue = `query(node('objectReturned',f(object('a',call(get('get'),'objectValue')))))`;
+  //
+  //   await expect(ask(queryForAnObjectValue)).resolves.toStrictEqual(
+  //     expectedObjectValue
+  //   );
+  // });
```

Comment:
Instead of having commented code here can we link to a GitHub issue in a comment and maybe have that code there?

Reply:
@YonatanKra I am fine with leaving the code there, but please please add a link to the GitHub issue.

Reply:
I don't like comments in code - especially when we have git :)
So I've copied the code to the issue itself.